Source/RomanNumeral.c

```diff
@@ -6,11 +6,41 @@
 #include <stdio.h>
 #include "RomanNumeral.h"
 
-static char * Append(char *output, const char *toAppend)
+static char *Append(char *output, const char *toAppend)
 {
    return output + sprintf(output, "%s", toAppend);
 }
 
+typedef struct
+{
+   int num;
+   const char *romanNumeral;
+} TableEntry_t;
+
+static const TableEntry_t romanNumeralTable[] = {
+   { 1000, "M" },
+   { 900, "CM" },
+   { 500, "D" },
+   { 400, "CD" },
+   { 100, "C" },
+   { 90, "XC" },
+   { 50, "L" },
+   { 40, "XL" },
+   { 10, "X" },
+   { 9, "IX" },
+   { 5, "V" },
+   { 4, "IV" },
+   { 1, "I" }
+};
+
 void RomanNumeral_Convert(unsigned input, char *output)
 {
+   for(int i = 0; i < sizeof(romanNumeralTable) / sizeof(romanNumeralTable[0]); i++)
```

Comment:
You'll find this useful and much more readable: https://github.com/geappliances/applcommon.applcommon/blob/master/Utilities/utils.h#L24

Reply:
Should I define it myself in the c file, or import that header itself?

Reply:
Future reference for when you are working.